### DISCUSSION TITLE
Failed to compile -  Error: Can't resolve '...' .bg-

### DISCUSSION THREAD

**Discussion by facos86** [tailwindlabs/tailwindcss#8503] [Help] [answered]
**What version of Tailwind CSS are you using?**
    "react": "^17.0.2",
   "cross-env": "7.0.3",
    "cssnano": "5.1.10",
    "postcss-cli": "9.1.0",
    "react-error-overlay": "^6.0.9",
    "release-it": "13.6.4",
    "tailwindcss": "^3.0.24",
    "autoprefixer": "^10.4.7"

**Describe your issue**

Hi,
I am encountering a problem with the latest version of tailwindcss.
It generates "background-image: url('...');" in tailwind.output.css, the three dots '...' cannot be resolved.
<img width="589" alt="Screenshot 2022-06-02 at 18 29 37" src="https://user-images.githubusercontent.com/2879114/171679351-34d14070-2a20-45a9-95a2-4b5d9ca3856e.png">
<img width="1392" alt="Screenshot 2022-06-02 at 18 31 39" src="https://user-images.githubusercontent.com/2879114/171679406-7ff5fc2d-ff75-4565-8da1-b7ce808ccda7.png">

Is there any configuration issues or is there a bug with my current stup ?

**Comment by adamwathan** [comment, marked as answer]
Hey! Those look like classes being generated based on Tailwind's test suite, which makes me think your `content` configuration is watching `node_modules` when it shouldn't be.

https://tailwindcss.com/docs/content-configuration#pattern-recommendations

**Reply by facos86** [reply]
You are absolutely right, the regexp in "content" was missconfigured.
Thanks for the quick response!

**Reply by ZeeshanAhmadKhalil** [reply]
I am getting this error:

```
Module not found: Can't resolve './${selectedImagePath}'

Import trace for requested module:
./styles/globals.css
./pages/_app.js
```

because I used variable inside the tailwind class as follows

`bg-[url('${selectedImagePath}')]`

Now i am unable to reset my output file `globals.css` and the error persists.

**Reply by tonyabracadabra** [reply]
running into the same issue, is there any update on this?

**Reply by wongjn** [reply]
You cannot use string interpolation in class names, [as per the documentation](https://tailwindcss.com/docs/content-configuration#dynamic-class-names):
> The most important implication of how Tailwind extracts class names is that it will only find classes that exist as complete unbroken strings in your source files.
>
> If you use string interpolation or concatenate partial class names together, Tailwind will not find them and therefore will not generate the corresponding CSS

**Reply by Drex72** [reply]
Were you able to solve this?

**Reply by boblamceo** [reply]
@ZeeshanAhmadKhalil same here

**Reply by tomBoyd9002** [reply]
> I am getting this error:
> 
> ```
> Module not found: Can't resolve './${selectedImagePath}'
> 
> Import trace for requested module:
> ./styles/globals.css
> ./pages/_app.js
> ```
> 
> because I used variable inside the tailwind class as follows
> 
> `bg-[url('${selectedImagePath}')]`
> 
> Now i am unable to reset my output file `globals.css` and the error persists.

delete the 

> I am getting this error:
> 
> ```
> Module not found: Can't resolve './${selectedImagePath}'
> 
> Import trace for requested module:
> ./styles/globals.css
> ./pages/_app.js
> ```
> 
> because I used variable inside the tailwind class as follows
> 
> `bg-[url('${selectedImagePath}')]`
> 
> Now i am unable to reset my output file `globals.css` and the error persists.

Delete the bg variable save and restart the server, then put a working bg variable in, then save and refresh the server, it will reset your global css. I have no clue how to make it display properly sadly.

**Reply by ZeeshanAhmadKhalil** [reply]
@boblamceo  i found an old `yarn.lock` file which made it run. I was unable to resolve the above error which I get when generating new `yarn.lock` file and then implementing the above solution.

**Reply by kenny-67** [reply]
@tomBoyd9002 How were you able to resolve your issue?
